### PR TITLE
feat: Initial pass using Github Actions

### PR DESCRIPTION
**Pull request by thecristen**
Asana ticket: [Migration OpenTripPlanner CI to Github Actions](https://app.asana.com/0/385363666817452/1199946732081254/f)

Two main tasks:

- Run tests -  this works! Take a look at this PR's checks and `tests.yml`.
- Deploy to dev - WIP

### CODE REVIEW

**Comment by thecristen** [comment]
@mkennedm I was trying to request a review from someone on @mbta/trc and it randomly chose you! 😅 Please feel free to disregard if unable to review this workflow.

File: .github/workflows/deploy-dev.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: ${{ github.workspace }}/OTP
+
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: 8
+          settings-path: ${{ github.workspace }}/OTP/mbta/semaphore
```
Comment (lemald) [inline .github/workflows/deploy-dev.yml:26]: Just curious, did you try using ASDF instead? I don't actually know enough about Java to say if that would have any concrete advantages or disadvantages over this approach.

Comment (lemald) [reply .github/workflows/deploy-dev.yml:26]: Also regarding `settings-path`, should we rename that directory to something that doesn't mention Semaphore?

File: mbta/semaphore/custom-settings.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<settings xmlns="http://maven.apache.org/SETTINGS/1.2.0"
```
Comment (lemald) [inline mbta/semaphore/custom-settings.xml:1]: Just curious - why did you need to create this file? Was it auto-generated by some tool or did you have to make it yourself?

Comment (thecristen) [reply mbta/semaphore/custom-settings.xml:1]: I had made it myself because I kept getting build errors (both here and also locally) because Maven didn't like insecure URLs (see "Blocked mirror for repositories" [in the error here for example](https://github.com/mbta/OpenTripPlanner/runs/2923312837?check_suite_focus=true)). I think that's a relatively new "feature" of Maven. Do these errors not happen to other folks? 🤔

File: .github/workflows/deploy-dev.yml (hunk 1 above)
Comment (lemald) [reply .github/workflows/deploy-dev.yml:26]: I suppose the advantage of `actions/setup-java` is that it's presumably written by people who know more about Java than we do, but on the other hand if we take this approach that's one more little spot in the repo we'd have to update if we changed anything about the Java version going forward.

File: mbta/semaphore/custom-settings.xml (hunk 2 above)
Comment (lemald) [reply mbta/semaphore/custom-settings.xml:1]: Huh, weird, I had never run into that before.

Comment (thecristen) [reply mbta/semaphore/custom-settings.xml:1]: I wonder if there's some other configuration I'm missing.

File: .github/workflows/deploy-dev.yml (hunk 1 above)
Comment (thecristen) [reply .github/workflows/deploy-dev.yml:26]: The `asdf-vm` family of actions aren't set up to handle subdirectories (and this action is a bit unconventional, loading OTP into one folder and the onebusaway modules into a sibling directory), so I'm not sure I'd be able to get ASDF to work easily for this workflow anyway. Would you be OK leaving these actions to use `actions/setup-java` for now, and update them to use ASDF at a future point?

Comment (lemald) [reply .github/workflows/deploy-dev.yml:26]: Yeah that's fine by me!

**Review by lemald — APPROVED** [review]
👍

**Comment by thecristen** [comment]
Thanks, might follow up with another PR depending how the deploy works! 👍🏼
